src/View/Helper/FormHelper.php

```diff
@@ -1345,7 +1345,7 @@ protected function _magicOptions(string $fieldName, array $options, bool $allowO
 
             if ($this->getConfig('autoSetCustomValidity')) {
                 $options['oninvalid'] = "this.setCustomValidity(''); "
-                    . "if (!this.validity.valid) this.setCustomValidity('$message')";
+                    . "if (!this.value) this.setCustomValidity('$message')";
```

Comment:
Will this still take HTML5 pattern validations into account?

Reply:
I updated the fiddle. As for the email field the custom validity message is shown only when field is empty. When failing to match the pattern it shows bower's default message.